### ISSUE TITLE
java.lang.IllegalAccessError on registration

### ISSUE THREAD

**Issue by bermudalocket** [JorelAli/1.13-Command-API#128]
**CommandAPI version**
3.4

**Minecraft version**
1.16.1

**Describe the bug**
The following exception is thrown when my plugin attempts to register a command at startup.
```
2020-08-17 17:06:49 | [RAW] java.lang.IllegalAccessError: class dev.jorel.commandapi.CommandAPICommand tried to access method 'void dev.jorel.commandapi.CommandAPI.register(java.lang.String, dev.jorel.commandapi.CommandPermission, java.lang.String[], java.util.LinkedHashMap, dev.jorel.commandapi.CustomCommandExecutor)' (dev.jorel.commandapi.CommandAPICommand is in unnamed module of loader org.bukkit.plugin.java.PluginClassLoader @656a8d3c; dev.jorel.commandapi.CommandAPI is in unnamed module of loader org.bukkit.plugin.java.PluginClassLoader @4a38f21d)
2020-08-17 17:06:49 | [RAW] 	at dev.jorel.commandapi.CommandAPICommand.register(CommandAPICommand.java:208) ~[?:?]
```

**My code**
Your sample code reproduces the behavior.
```java
LinkedHashMap<String, Argument> args = new LinkedHashMap<>();
CommandAPI.getInstance().register("mycmd", args, (sender, args) -> {
    //blah
});
```

**Expected behavior**
The exception not to be thrown, the command to successfully register.

**Comment by JorelAli** [comment]
You're using the wrong code - that was the old registration method (pre version 3.0).

The new registration method is defined as stated in [the documentation](https://www.jorel.dev/1.13-Command-API/3.4/commandregistration.html):

```java
new CommandAPICommand("mycmd")
    .withArguments(args)
    .executes((sender, a) -> ...)
    .register();
```

**Comment by JorelAli** [comment]
Can confirm there are a few old parts of the documentation that have not been updated to the new syntax. I'll make sure this is fixed for version 4.0 due to release later this week
